lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -17,15 +17,44 @@ class AwsCompileFunctions {
     };
   }
 
-  compileRole(role) {
-    if (typeof role === 'object') {
-      // role is an "Fn::GetAtt" object
-      return role;
-    } else if (role.indexOf(':') === -1) {
-      // role is a Logical Role Name
-      return { 'Fn::GetAtt': [role, 'Arn'] };
+  compileRole(newFunction, role) {
+    const compiledFunction = newFunction;
+    const unnsupportedRoleError = new this.serverless.classes
+      .Error(`Unsupported role provided: "${role}"`);
```

Comment:
Would be great if this resolves the corresponding type.

When I put in e.g. this:

```
role:
  Foo: bar
```

I get the error `Unsupported role provided: "[object Object]"`. A stringified version of the object would be awesome so that I know which one breaks.